src/coreclr/src/pal/src/thread/thread.cpp

```diff
@@ -1641,6 +1641,13 @@ CorUnix::InternalSetThreadDescription(
 
     pTargetThread->Lock(pThread);
 
+    // Ignore requests to set the main thread name because
```

Comment:
I put this in pal, because it is a good place for calling `getpid`.

1. I'm not sure whether this needs to be under `->Lock`.
2. When looking at the code above `->Lock` I noticed `goto InternalSetThreadDescriptionExit;` is also used, though I don't understand how it knows when it shouldn't `->Unlock`.

Reply:
> I don't understand how it knows when it shouldn't ->Unlock.

This may be a bug. `InternalLeaveCriticalSection` would no-op (I think) so it would not be noticed unless we followed this codepath with a checked runtime so the asserts fired?

Reply:
I looked into it a bit more. It is not a bug: `pTargetThread` will be `null`, and `->Unlock` isn't called.